include/swift/AST/Decl.h

```diff
@@ -3333,7 +3333,7 @@ class ProtocolDecl : public NominalTypeDecl {
 
   ArrayRef<ProtocolDecl *> InheritedProtocols;
 
-  llvm::DenseMap<ValueDecl *, ConcreteDeclRef> DefaultWitnesses;
+  llvm::DenseMap<ValueDecl *, ValueDecl *> DefaultWitnesses;
```

Comment:
Don't default witnesses require substitutions sometimes?

Imagine the requirement looks like:

    protocol P { func foo(x: Int) }

And the default witness is

    extension P { func foo<T>(x: T) { ... } }